app/common/elasticsearchclient.py

```diff
@@ -240,19 +240,25 @@ def best_hit_search(self,
         # current_app.logger.debug("Got %d Hits in %ims" % (res['hits']['total'], res['took']))
         data = []
         
-        for res in results['responses']:
-            res_data = [];
-            for hit in res['hits']['hits']:
+        #there are len(params.q) responses - one per query
+        for i in range(len(results['responses'])):
+            name = params.q[i]  #even though we are guaranteed that responses come back in order, and can match query to the result - this might be convenient to have       
+            res = results['responses'][i]
+            if(len(res['hits']['hits'])>0):
+                hit = res['hits']['hits'][0] #expect either 1 result or none
                 highlight = ''
                 if 'highlight' in hit:
                     highlight = hit['highlight']
                 datapoint = dict(type=hit['_type'],
                                  data=hit['_source'],
                                  id=hit['_id'],
                                  score=hit['_score'],
-                                 highlight=highlight)
-                res_data.append(datapoint)
-            data.append(res_data)
+                                 highlight=highlight,
+                                 q=name)
+            else:
+                datapoint = dict(id='',q=name)
```

Comment:
if you need to pass the id field here, then it would be better to use `None` instead of an empty string to signal that nothing was found. We could also specifically state a `"not found"` status here.